lib/fog/aws/requests/dynamodb/query.rb

```diff
@@ -25,7 +23,17 @@ class Real
         #     * 'Count'<~Integer> - number of items in response
         #     * 'Items'<~Array> - array of items returned
         #     * 'LastEvaluatedKey'<~Hash> - last key scanned, can be passed to ExclusiveStartKey for pagination
-        def query(table_name, hash_key, options = {})
+        #
+        # See DynamoDB Documentation: http://docs.aws.amazon.com/amazondynamodb/latest/APIReference/API_Query.html
+        #
+        def query(table_name, options = {}, hash_key_deprecated = nil)
+          if has_deprecated_hash_key || (options.keys.length == 1 && [:S, :N, :B].include?(options.keys.first.to_sym))
```

Comment:
It is a method argument. We need it because the method signature changes, as DynamoDB doesn't have any `HashKey` option anymore. We need to do a string `KeyConditionExpression` instead.